src/analyses/variable-sensitivity/value_set_abstract_value.cpp

```diff
@@ -125,3 +126,92 @@ void value_set_abstract_valuet::output(
   }
   out << "}";
 }
+
+static void merge_all_possible_results(
+  std::shared_ptr<const value_set_abstract_valuet> &out_value,
+  const exprt &expr,
+  const std::vector<const value_set_abstract_valuet *> &operand_value_sets,
+  const namespacet &ns,
+  std::vector<exprt> &operands_so_far)
+{
+  if(expr.operands().size() == operands_so_far.size())
+  {
+    exprt expr_with_evaluated_operands_filled_in = expr;
+    expr_with_evaluated_operands_filled_in.operands() = operands_so_far;
+    simplify(expr_with_evaluated_operands_filled_in, ns);
+    if(expr_with_evaluated_operands_filled_in.is_constant())
+    {
+      bool out_modifications;
+      auto post_merge = abstract_objectt::merge(
+        out_value,
+        std::make_shared<value_set_abstract_valuet>(
+          expr.type(),
+          value_set_abstract_valuet::valuest{
+            expr_with_evaluated_operands_filled_in}),
+        out_modifications);
```

Comment:
For very large expressions there might be some value in exiting early here when value sets get too large and just become TOP.